common/djangoapps/xblock_django/user_service.py

```diff
@@ -28,12 +28,17 @@ def __init__(self, django_user, **kwargs):
         self._django_user = django_user
         if self._django_user:
             self._django_user.user_is_staff = kwargs.get('user_is_staff', False)
+        self._xblock_user = None
 
     def get_current_user(self):
         """
         Returns the currently-logged in user, as an instance of XBlockUser
+
+        XBlockUser is cached after the first call.
         """
-        return self._convert_django_user_to_xblock_user(self._django_user)
+        if not self._xblock_user:
+            self._xblock_user = self._convert_django_user_to_xblock_user(self._django_user)
```

Comment:
Is it ok to cache this XBlockUser?

Reply:
We can do it if some data is being fetch from the db but we do not expect that data to change between invocations during the life-time of the object. 

https://docs.python.org/3/library/functools.html#functools.cached_property is the official way if we need to do it.

Reply:
Ok, I thought the [user preferences might be hitting the db](https://github.com/edx/edx-platform/blob/3152c14192dcccac79b7b7a76eba91b380eba78a/common/djangoapps/xblock_django/user_service.py#L89), but they are [likely prefetched](https://github.com/edx/edx-platform/blob/3152c14192dcccac79b7b7a76eba91b380eba78a/openedx/core/djangoapps/user_api/views.py#L36). Will revert these changes, and we can use `@cached_property` if the query count goes up.